### DISCUSSION TITLE
How matches are presented and selected

### DISCUSSION THREAD

**Discussion by Oshyan** [PullDating/Org#23] [Ideas & Features]
Two of the key mechanics of any dating app are:
1. How do people find/access matches, i.e. how are they "presented" or made available to the user
2. How do people select or otherwise indicate interest in a match

These two things together form a significant part of the overall user experience of the app, and even more importantly they affect how people engage with and feel about both the app, and the people/profiles within it. We need to carefully consider the best approach(es) to these questions, with our core goals in mind. We want to promote positive interaction, good matches that align with the user's own goals, etc. That fact alone (that the user's goals may vary) might also suggest *different* mechanics depending on what an individual user's goals are, e.g. swiping through an infinite, sequential feed might be demonstrably better for finding matches if someone wants a hookup, whereas searching with filters and selecting individual matches from a gallery, or based on question responses, etc. might be better for those seeking other kinds of matches. All this needs to be taken into account.

There are a couple of major examples of existing paradigms in current dating apps that I think should be considered.

# Accessing Matches
This is basically how matches are presented to a user. The actual consideration of how matches are made, i.e. based on what *criteria*, is separate (but related, of course).
## Sequential
This is the most common approach in modern dating apps, exemplified by Tinder, and most often tied with (but not limited to) a single stream of potential matches. Again the back-end algorithms that determine who gets *into* the feed (matching algorithms) are another consideration. A sequential approach essentially requires people to definitively decide on each profile, yes or no, before they can see another one.

There are interesting pros and cons to this that might benefit from some research/testing on actual daters (perhaps some of that research has been done). For example personally I want to have more of a gallery view (see below), but this can also result in inaction, or focusing too much on the very top most aesthetic people, while missing out on a bunch of otherwise good matches that *if* one were forced to choose, they might actually match with and discover are great. It turns out, research-wise, that aesthetic, physical attractiveness doesn't actually make anyone happier in relationship, for the most part, so pushing back against this impulse may be beneficial, as long as it's done in the right ways.
## Gallery or List
One of the most user-empowering ways to present potential matches is with a full gallery or list view that the user can browse freely and simply click on any profile that catches their eye. There are ideas here that might interact in special ways with this, e.g. #19 VibeCheck (in which case maybe a blurred/pixelated photo?), but fundamentally I tend to prefer this model. OKCupid of old allowed this, but no longer, I think. It does show 1-way matches this way though (if you are a paying user), i.e. people who have already indicated an interest in you (see below).
## Search
Most modern dating apps don't give you good tools to do intentional searches with, but my personal opinion and instinct is that it has strong benefits to consider in terms of user agency, etc. Of course subconsciously many people might be "looking" for the wrong things, and so we may have to be thinking on a further meta level about the difference between what people might search for, and who might actually be good matches for them, as well as the difference between what someone is looking for, and whether others are open to what that searcher offers (e.g. someone is open to anyone in a 500 mi radius because they're fine with long-distance, but many others aren't). At any rate the core Search mechanic is what is the primary consideration here. Search filters, keywords, question responses, etc.

## Finding Matches vs. Evaluating Interest
Another dimension to this is that there can (and perhaps should) be a distinction between how profiles/potential matches are presented when you are actively searching for new people vs. viewing people who have already indicated interest in you (let's call them 1-way matches). Many current dating apps have a sequential stream for 1, and a "gallery" for the other (even if the "gallery" is a paid feature, e.g. Hinge). 

### Matches in Search Stream
It's also an open question whether existing 1-way potential matches (other person has indicated interest) should just all go in one place where the user can access them any time, and thus when they are searching they only see *new* profiles, or whether - like most current dating apps - 1-way matches should be in the search/new stream. In a *non-profit* dating app, where you don't have incentive to keep people swiping and thus withhold information, arguably there should be a clear distinction between these two and the user can choose to access whichever they want. But there may be more subtle psychological arguments in favor of having that and *also* putting those 1-way matches into the stream.

# Selecting Matches
This encompasses another whole range of things. There is the simple part, the difference between swipe left/right and clicking buttons for yes/no, which seems less consequential. But there are also things like "Match + Intro" to consider, as well as different degrees of indicating interest, or perhaps even different *types* of interest one could explicitly indicate upon matching (e.g. "Interested for hookup" (where the other user's profile also allows that kind of interest) vs. "Interested for relationship", etc. 

I have more thoughts on this, but will end here for now and try to add more later.
# How we decide
I think we need to advocate for our choice of approach, among these, or suggest alternatives if we have any. Discussion should focus around compelling and specific arguments as to why we believe a given approach will result in a superior outcome, aligned with our goals (finding people the best matches for their own personal relationship goals, etc.). And keep in mind that achieving the best outcome is not always intuitive. In some cases we may even have to consider choices between what seems most open, fair, or desired by the user community, and what we can actually see (based on some evidence, research, etc.) may in fact be more *effective* for achieving the goals. Regardless, the more well argued our respective choices of approach, the better we can evaluate and decide.

## Multiple options or "modes"
One way of handling the decision, if there is no clear winner for us, is to offer multiple options, based on some criteria, or just free user choice, etc. Of course narrowing it down helps reduce the functions that need to be developed, so there is value in it, and probably for other reasons too. But this is an option.
## Remaining experimental and flexible
Ultimately whatever we decide to implement may not turn out to work as we hope in practice. So we'll remain open to adjusting or changing the dynamic entirely, if needed. But ideally we'll have metrics and testable hypotheses to base such changes on, and not just intuition or guess work. The arguments we make for each approach would ideally include such testable assertions/hypotheses.

**Comment by aklidata** [comment]
### Note about research
------------------------------------------
I noticed on r/tinder these "surveys" that people run of their dating experience that yield data on how their dating experience went, what the result from it was etc... 
We could compile that data or find pre-compiled data to base our preliminary research on it.

**Comment by ajbott** [comment]
There are a lot of good things to think about throughout. I think the core thing to think about is our approach to be as dynamic as possible and be as open to trying out as many ideas as we can to see what works. In order to do this, I think user freedom to be able to experiment with various options to grant them the power that aligns with their own dating app interests.
